sdks/python/apache_beam/io/gcp/tests/utils.py

```diff
@@ -41,17 +46,16 @@ def delete_bq_table(project, dataset, table):
     table:   Name of the table.
   """
   logging.info('Clean up a Bigquery table with project: %s, dataset: %s, '
-               'table: %s', project, dataset, table)
+               'table: %s.', project, dataset, table)
   bq_dataset = bigquery.Client(project=project).dataset(dataset)
   if not bq_dataset.exists():
-    logging.warning('Delete failed. Bigquery dataset %s doesn\'t exist in '
-                    'project %s.', dataset, project)
-    return
+    raise GcpIOError('Failed to cleanup. Bigquery dataset %s doesn\'t exist '
+                     'in project %s.' % dataset, project)
   bq_table = bq_dataset.table(table)
   if not bq_table.exists():
-    logging.warning('Delete failed. Biqeury table %s doesn\'t exist in '
-                    'project %s, dataset %s', table, project, dataset)
-    return
+    raise GcpIOError('Failed to cleanup. Biqeury table %s doesn\'t exist '
+                     'in project %s, dataset %s.' % table, project, dataset)
   bq_table.delete()
   if bq_table.exists():
-    raise RuntimeError('Delete failed. Bigquery table %s still exists' % table)
+    raise RuntimeError('Failed to cleanup. Bigquery table %s still exists '
```

Comment:
If table exists after delete due to Bigquery inconsistent, `GcpTestIOError` will raise probably in next retry and test will fail. If the problem is in `bq_table.delete()` but no error raise, I think retry is also helpful.